src/core/shape_inference/include/slice_shape_inference.hpp

```diff
@@ -0,0 +1,153 @@
+                              "` input must be a 1D tensor. Got rank: ",
+                              shape_rank);
+
+        if (input_rank.is_static()) {
+            NODE_VALIDATION_CHECK(op,
+                                  shape_rank.is_dynamic() || shape[0].get_min_length() <= input_rank.get_length(),
+                                  "Slice `",
+                                  slice::shape_names[i - 1],
+                                  "` input dim size can't be bigger than `data` rank.");
+        }
+    }
+
+    const auto& start_shape = input_shapes[1];
+    const auto& stop_shape = input_shapes[2];
+    const auto& step_shape = input_shapes[3];
+
+    NODE_VALIDATION_CHECK(
+        op,
+        start_shape.compatible(stop_shape) && start_shape.compatible(step_shape) && stop_shape.compatible(step_shape),
+        "Slice `start`, `stop`, `step` inputs must have compatible shapes.");
+
+    // it is not possible to define output shape if input data shape rank is undefined
+    // even the lengths of begin, end, or strides are defined
+    if (input_rank.is_dynamic()) {
+        output_shapes[0] = PartialShape::dynamic();
+        return;
+    }
+
+    // compute constant values of begin, end, and strides if possible
+    const auto start = slice::get_input_bounds<T>(op, 1, constant_data);
+    const auto stop = slice::get_input_bounds<T>(op, 2, constant_data);
+    const auto steps = get_input_const_data_as<T, int64_t>(op, 3, constant_data);
+
+    slice::AxesMap axes_map;
+    if (input_shapes.size() > 4) {
+        NODE_VALIDATION_CHECK(op,
+                              input_shapes[4].compatible(start_shape),
+                              "Slice `axes` input must have compatible shape with `start`, `stop`, `step` inputs.");
+
+        if (auto axes = get_input_const_data_as<T, int64_t>(op, 4, constant_data)) {
+            ov::normalize_axes(op, input_shape.rank().get_length(), *axes);
+            axes_map.add(*axes);
+            NODE_VALIDATION_CHECK(op, axes_map.is_valid, "Slice values in `axes` input must be unique.");
+        }
+    } else if (start) {
+        axes_map.generate_n(start->size());
+    }
+
+    auto axis_it = axes_map.m.cbegin();
+
+    auto& out = output_shapes.front();
+    out.resize(0);
+    out.reserve(input_shape.size());
+    for (size_t dim_idx = 0; dim_idx < input_shape.size(); ++dim_idx) {
+        const DimType& input_dim = input_shape[dim_idx];
+
+        if (axes_map.is_valid && (axis_it != axes_map.m.cend()) && (axis_it->first == dim_idx)) {
+            const auto& i = axis_it->second;
+
+            if (start && stop && steps) {
+                const auto& step = (*steps)[i];
+                NODE_VALIDATION_CHECK(op, step != 0, "Step must be non-zero");
+                out.push_back(slice::make_dim(input_dim, (*start)[i], (*stop)[i], step));
+            } else {
+                out.emplace_back(0, input_dim.get_max_length());
+            }
```

Comment:
could we calculate more precisely as we did for strided slice? for example if stop is known we can use that to define out dimension more precisely

Reply:
So I'm not sure when only stop is known that is possible to get more precise result, e.g.

Input shape is {20} and stop is 5 then possible output could be:
-  {0,5}   when assumes that start is begin of dimension and step is positive
-  {0,15} when assumes that start is end of dimension and step is negative

The strided slice also do sliced dimension calculations only when got start, stop, step (checked version before review of shape inference) and in case when values are not know the result is for shape{20} -> {0,20} (slice give same result)